service/controller/v3/resource/dnsrecord/resource.go

```diff
@@ -45,20 +48,29 @@ func (r *Resource) Name() string {
 	return Name
 }
 
-func (r *Resource) getDNSRecordSetsClient() (*dns.RecordSetsClient, error) {
-	azureClients, err := client.NewAzureClientSet(r.azureConfig)
+func (r *Resource) getDNSRecordSetsHostClient() (*dns.RecordSetsClient, error) {
+	azureClients, err := client.NewAzureClientSet(r.hostAzureConfig)
 	if err != nil {
 		return nil, microerror.Mask(err)
 	}
 
 	return azureClients.DNSRecordSetsClient, nil
 }
 
-func (r *Resource) getDNSZonesClient() (*dns.ZonesClient, error) {
-	azureClients, err := client.NewAzureClientSet(r.azureConfig)
+func (r *Resource) getDNSRecordSetsGuestClient(ctx context.Context) (*dns.RecordSetsClient, error) {
+	sc, err := controllercontext.FromContext(ctx)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	return sc.AzureClientSet.DNSRecordSetsClient, nil
+}
+
+func (r *Resource) getDNSZonesClient(ctx context.Context) (*dns.ZonesClient, error) {
```

Comment:
Can you rename that one to `getDNSZonesGuestClient` to stay consistent?

Reply:
done